server/rest/src/main/java/org/apache/karaf/cave/server/rest/Repository.java

```diff
@@ -16,21 +16,145 @@
  */
 package org.apache.karaf.cave.server.rest;
 
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
 import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.XmlType;
 
 import org.apache.karaf.cave.server.api.CaveRepository;
+import org.apache.karaf.util.XmlUtils;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
 
 @XmlRootElement(name = "cave-repository")
+@XmlAccessorType(XmlAccessType.NONE)
+@XmlType(name = "Repository", propOrder = {"name", "increment", "location", "resources"})
 public class Repository {
 
-    private final CaveRepository repository;
```

Comment:
How do you access the actual `CaveRepository` now (I don't find the location) ?

Reply:
I don't change the implementation of the method in the Service class :
```
public Repository getRepository(@PathParam("name") String name) {
        CaveRepository repository = service.getRepository(name);
        if (repository != null) {
            return new Repository(repository);
        } else {
            return null;
        }
}
```
repository service is passed in the constructor and used directly the retreive repository data :
```
public Repository(CaveRepository repository) {
        this.name = repository.getName();
        this.increment = repository.getIncrement();
        this.location = repository.getLocation();

        try {
Document doc = XmlUtils.parse(repository.getRepositoryXml().toExternalForm());
...
```

Reply:
Got it now. Thanks !